unstructured/ingest/enhanced_dataclass/core.py

```diff
@@ -0,0 +1,197 @@
+import _thread
+import copy
+import functools
+import warnings
+from dataclasses import fields
+
+from dataclasses_json.core import (
+    MISSING,
+    Collection,
+    Enum,
+    Mapping,
+    _decode_generic,
+    _decode_letter_case_overrides,
+    _encode_overrides,
+    _handle_undefined_parameters_safe,
+    _is_new_type,
+    _is_optional,
+    _is_supported_generic,
+    _isinstance_safe,
+    _support_extended_types,
+    _user_overrides_or_exts,
+    get_type_hints,
+    is_dataclass,
+)
+
+
+def _recursive_repr(user_function):
+    # Copied from dataclasses as this method isn't exposed for importing
```

Comment:
```suggestion
    """Decorator to make a repr function return "..." for a recursive call. Useful in cases where calling __repr__ would cause recursion errors. """
    # Copied from dataclasses as this method isn't exposed for importing
```